raft/log.go

```diff
@@ -211,20 +216,26 @@ func (l *raftLog) term(i uint64) uint64 {
 		return t
 	}
 	if err == ErrCompacted {
-		return 0
+		return math.MaxUint64
```

Comment:
the problem here is: 0 is actual a valid term. The term of the dummy entry is 0. I would love to add a errCompacted to this function. But this also means we need to deal with the error at line207 together.